### ISSUE TITLE
Using LAN adapter instead of RS

### ISSUE THREAD

**Issue by nikowitt** [assembly12/Foxess-T-series-ESPHome-Home-Assistant#6]
Hi there,
in https://github.com/StealthChesnut/HA-FoxESS-Modbus, it is mentioned that the LAN adapter will also work to get the data. Does this also work in this case?
Regards,
Niko

**Comment by assembly12** [comment]
Yes, it's possible. It will however not be compatible with this github repository.
[https://community.home-assistant.io/t/foxess-inverter-modbus/459313/48](https://community.home-assistant.io/t/foxess-inverter-modbus/459313/48)

**Comment by nikowitt** [comment]
I see, need to have a closer look. Are there place on your end to make this possible for WLAN, so basically providing the capabilities like in https://github.com/StealthChesnut/HA-FoxESS-Modbus for the T series?

**Comment by assembly12** [comment]
Hello!
Im not sure if this is a misunderstanding on my part.
In your opening post you asked about LAN and in your recent post you ask about WLAN; is there a typo or autocorrection?

All the code in this component runs on the ESP32, which in turn connects to HA via WLAN (wifi).
Is there a specific reason you don't want to use an ESP-based solution?
There is some soldering involved, but apart from that it's really just copy paste.

**Comment by nikowitt** [comment]
I am sorry, "WLAN" was a mistake. I am talking about the LAN adapter as everything is suppose to go over a wired connection. As I am at the beginning of the journey, I am not really familiar with the connectivity of devices yet. From the other repo, I was under the impression that in general, using LAN works without additional hardware but with maybe only some different mapping required.

So in a nutshell, I am looking for an easy way to connect a T series inverter via LAN adapter to HA to add as little additional hardware (and complexity to the whole system) as possible :-)

**Comment by assembly12** [comment]
Ok. Well, this component works with an ESP32/ESP8266-board which usally works with wifi. There are ESP modules with a build in LAN port, so theoretically you might be able to get a wired solution working; i've never used one up till now however.

Since the Foxess inverter only has a RS485 port and no LAN port you will be needing additional hardware whichever route you decide to go. Let me just list a couple of characteristics for both solutions, maybe it will help you in making a choice.

**With a LAN adapter:**
- almost no tinkering with hardware, basically just plug in the connectors
- only one additional hardware component needed
- everything has to be setup in node-red, which requires understanding the Foxess protocol

**With an ESP-based solution (either WLAN, or LAN):**
- some amount of soldering needed (only on  the additional hardware components, connection to the inverter is by plug)
- a couple of different hardware components are needed (see the parts list)
- basically no tinkering with software, just copy paste and it should work

The cost for both solutions is probably about the same, give or take a couple of Dollars.
In short; the LAN adapter requires you to do more "software stuff" and the ESP solution requires you to do more "hardware stuff" (soldering).

Hope this makes the difference between both methods more clear!

**Comment by nikowitt** [comment]
Sorry for the delay. Thanks for the clarification! I see, I was also under the wrong impression that the delivered datalogger (Smart LAN) that has a LAN port (attached via USB to the inverter) is sufficient, but that also seems to be wrong. Thanks for the clarification, I will start having a look at appropriate hardware.
